Core/Config.cpp

```diff
@@ -1529,6 +1529,8 @@ void Config::RemoveRecent(const std::string &file) {
 	}
 }
 
+// TODO: This can be called during shutdown at a point where we have no NativeActivity object.
```

Comment:
We should make `ExistsFast()` always return true for content URIs when it can't open them.  It could just be a simple proxy file loader (like `RetryingFileLoader.`)  The model I'd use would be:

```c++
class ContentURIFileLoader : public ProxiedFileLoader {
public:
	ContentURIFileLoader(const std::string &filename) {
		int fd = Android_OpenContentUriFd(filename);
		backend_ = new LocalFileLoader(fd, filename);
	}

	bool ExistsFast() override {
		if (!nativeActivity) {
			// Assume it does if we don't have a NativeActivity right now.
			return true;
		}
		return backend_->ExistsFast();
	}
}
```

This could be defined only in app-android.cpp and returned by the factory - would ensure removal doesn't happen incorrectly.  This is the purpose of the chained model for file loaders.

-[Unknown]